runtime/compiler/trj9/runtime/J9Profiler.cpp

```diff
@@ -2398,15 +2415,14 @@ TR_PersistentProfileInfo::getCurrent(TR::Compilation *comp)
 void
 TR_PersistentProfileInfo::incRefCount(TR_PersistentProfileInfo *info)
    {
+   TR_ASSERT_FATAL(info->_refCount > 0, "Increment called on profile info with no references");
    VM_AtomicSupport::add((uintptr_t*) &(info->_refCount), 1);
-   TR_ASSERT_FATAL(info->_refCount, "Increment resulted in reference count of 0");
+   TR_ASSERT_FATAL(info->_refCount > 0, "Increment resulted in negative or zero reference count");
```

Comment:
I was under the impression this was safe. The current thread has just incremented the count and so it must perform a decrement before it would be possible for the `refCount` to reach zero. I suppose with shared references between threads, this might be possible, although they should have synchronization to avoid it.

The condition is more to avoid overflow than anything else, so the condition has been updated accordingly.